sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceReaderHolder.scala

```diff
@@ -28,9 +28,9 @@ import org.apache.spark.sql.sources.v2.reader._
 trait DataSourceReaderHolder {
 
   /**
-   * The full output of the data source reader, without column pruning.
+   * The output of the data source reader, without column pruning.
    */
-  def fullOutput: Seq[AttributeReference]
```

Comment:
@cloud-fan This fixes the bug I spoke to you offline about. 
The target of this PR is only master, not 2.3. So if you want to have this fix in 2.3.0, please make a separate PR accordingly.

Reply:
If this PR has to be merged to 2.3.0 branch does it require more additional changes?